Sources/SwiftParser/Declarations.swift

```diff
@@ -1694,11 +1694,62 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+    func isFixity(_ modifier: RawDeclModifierSyntax) -> Bool {
+      switch modifier.name {
+      case .keyword(.prefix),
+        .keyword(.infix),
+        .keyword(.postfix):
+        return true
+      default:
+        return false
+      }
+    }
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena)
+
+    var fixity: RawTokenSyntax?
+    var leftInFixityModifier: RawUnexpectedNodesSyntax?
+
+    if let modifiers = attrs.modifiers?.elements,
+      let firstFixityIndex = modifiers.firstIndex(where: { isFixity($0) })
+    {
+      let fixityModifier = modifiers[firstFixityIndex]
+      fixity = fixityModifier.name
+
+      leftInFixityModifier = RawUnexpectedNodesSyntax(combining: fixityModifier.unexpectedBetweenNameAndDetail, RawUnexpectedNodesSyntax([fixityModifier.detail], arena: self.arena), fixityModifier.unexpectedAfterDetail, arena: self.arena)
+
+      let beforeFixity = Array(modifiers[0..<firstFixityIndex])
+
+      unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: unexpectedBeforeFixity, RawUnexpectedNodesSyntax(beforeFixity, arena: self.arena), fixityModifier.unexpectedBeforeName, arena: self.arena)
+    }
```

Comment:
If `modifiers.firstIndex(where: { isFixity($0) })` returns `nil`, we drop the modifiers. To avoid that, you need to add an `else` branch to the `if` statement that adds all modifiers to `unexpectedBeforeFixity or `unexpectedBeforeOperatorKeyword`. Which one you choose doesn’t really matter here because there are no present tokens between these two unexpected sets.

Reply:
Good catch! I also added a test case for this.